src/js/utils/pagination.js

```diff
@@ -18,24 +18,13 @@ export const usePagination = ({ data, page, step, ...rest }) => {
 
   const itemsBeginIndex = step * (activePage - 1);
   const itemsEndIndex = itemsBeginIndex + step;
-  const getCurrentItems = useCallback(
-    items => {
-      if (Array.isArray(items)) {
-        return items.length
-          ? items.slice(itemsBeginIndex, itemsEndIndex)
-          : undefined;
-      }
-      return items;
-    },
-    [itemsBeginIndex, itemsEndIndex],
-  );
-  const [currentItems, setCurrentItems] = useState(getCurrentItems(data));
 
-  useEffect(() => {
-    if (data) {
-      setCurrentItems(getCurrentItems(data));
+  const currentItems = useMemo(() => {
+    if (Array.isArray(data)) {
+      return data.length ? data.slice(itemsBeginIndex, itemsEndIndex) : [];
     }
-  }, [data, getCurrentItems, setCurrentItems]);
+    return data;
```

Comment:
Should this instead return `[]` when data isn't an array?

Reply:
Yes I think that would be a safe return. No errors were being caused because the only case this return was being called was when List provided `data` as `undefined`, but I think for consistency it would make sense for pagination to return an empty array for currentItems.